### ISSUE TITLE
Consider adding a "map" to core/app source code

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-app#694]
It can be confusing for newcomers to dive into our source code, having a very short high level map of the repo's organization could be useful.

Not entirely unlike this [example](https://github.com/quicksilver/Quicksilver/blob/main/The_Map.md), but perhaps even shorter and more visual

**Comment by HoytRen** [comment]
Yes, I feel it will be very useful. especially when we are migarating some parts between core and app. Currently I had reviewed some parts, but I don't have an overview, or I may be able to draw a chart about this.
